Tests/PlayerTests/Helpers/XCTestCase+Player.swift

```diff
@@ -17,4 +18,19 @@ extension XCTestCase {
 
 		XCTFail("Optimized wait: Condition was not met")
 	}
+	
+	func asyncOptimizedWait(timeout: Double = 10.0, step: Double = 0.1, until condition: @escaping () -> Bool) async {
+		var timer: Double = 0
+		while timer < timeout {
+			try? await Task.sleep(seconds: step)
```

Comment:
[question] is the sleep less flaky than the `XCTWaiter.wait`?
[question2] Have you revisited the default timeout we have been using? (To know if timeout can be shorter, I remember we had increased but maybe after improvements it is not needed so long anymore)

Reply:
I really can't say what they do under the hood. I would expect both to be the same with respect to flakiness.
But `Task.sleep` feels like a better Primitive for what we are doing. `XCTWaiter.wait` has extra functionality, as it also allows for the expectation to be fullfilled earlier, and you can check the result to know if it timed out or was fullfilled. And we don't really need any of that here.

Compared to `XCTWaiter.wait` outside `optimizedWait` with a fixed time, this is more robust because we can use bigger timeouts without generally affecting the test speed.

I don't think the default timeout matters much, because we check continuously to exit early.